onyo/lib/git.py

```diff
@@ -401,3 +404,85 @@ def check_ignore(self, ignore: Path, paths: list[Path]) -> list[Path]:
 
     # TODO: git check-ignore --no-index --stdin (or via command call)  ->  lazy, check GitRepo.files once. (Same invalidation)
     #       -> OnyoRepo would use it to maintain a ignore list from a (top-level .onyoignore)? .onyo/ignore ? Both?
+
+    def _parse_log_output(self, lines: list[str]) -> dict:
+        """Generate a dict from the output of ``git log`` for one commit.
+
+        Internal helper that gets a list of git-log-output lines,
+        that represent a single commit and generates a dictionary from it.
+        """
+        import datetime
+        import re
+        regex_author = re.compile(r"Author:\s+(?P<name>\b.*\b)\s+<(?P<email>[^\s]+)>$")
```

Comment:
Yes, you understand correctly.
Not sure it's worth it moving out, since this means it's always going to be compiled even if never used in an onyo call. But wanted to decouple from usage so it can easily be moved and the used expressions are in one place.

Reply:
> Not sure it's worth it moving out, since this means it's always going to be compiled even if never used in an onyo call

I think that depends where you put it. If it's something that gets imported (similar to consts) then it wouldn't be.

Just something to keep in mind. We'll see when we get there. And we have benchmarking now. :-D